### PR TITLE
Create Flutter demo template + CI

### PR DESCRIPTION
**Pull request by gruvw**
To be merged 2024_03_17 at 23:59 when Tech Stack decision poll ends :rocket:

Adds a simple flutter demo template presented to the team :computer: 

It is set up for android only by default :iphone: 
  
You can add local native support on your machine for running the app natively using the following command inside the repo: `flutter create --platforms=windows,macos,linux .` (use only the platform that you prefer).

**Note**: always run your code on the android simulator before pushing and try to use [act](https://github.com/nektos/act) to run `act push` before pushing to GitHub in order to avoid small correction commits.

**Note**: remember to use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) for our commit messages. Please take 30 seconds to write a comprehensible commit message for your team and also for your future self :smiling_face_with_tear: 

To start reviewing this PR locally, run the following:
```sh
git clone https://github.com/ProximaEPFL/proxima.git
cd proxima
git checkout flutter-setup
```

### CODE REVIEW

**Comment by CHOOSEIT** [comment]
LGTM !

**Review by CHOOSEIT — APPROVED** [review]
LGTM !

**Review by camillelnne — APPROVED** [review]
Approved

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Flutter SDK
+        uses: subosito/flutter-action@v2
+        with:
+          channel: stable
+          cache: true
```
Comment (JoachimFavre) [inline .github/workflows/tests.yaml:16]: Very good idea

**Review by JoachimFavre — COMMENTED** [review]
Nice work

**Review by yoannLafore — APPROVED** [review]
LGTM

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (gruvw) [reply .github/workflows/tests.yaml:16]: test comment worked perfectly :)

**Pull request merged**